src/metrics/promotions.ts

```diff
@@ -33,21 +35,62 @@ export class PromotionMetric implements Metric {
     this.totals = { q: 0, r: 0, b: 0, n: 0 };
 
     this.promotionMap = promotionMap as any;
+
+    this.movesAndGamesWithMaxQueenCount = [];
   }
 
   processGame(
     game: { move: PrettyMove; board: Piece[] }[],
     metadata?: string[]
   ) {
-    for (const { move, board } of game) {
+
+    // 2 queens to start with
+    let thisGameQueenCount = 2;
+
+    for (const { move } of game) {
       // TODO: we can use flags instead of includes('=)
 
       if (move.originalString.includes('=')) {
         // update maps
         this.promotionMap[move.uas][move.promotion]++;
+        
+        // increment queen count if a promotion occurs
+        if (move.promotion === "q") {
+          thisGameQueenCount++;
+        }
       }
```

Comment:
minor change but we can directly use the .promotion flag's existence

Reply:
wdym?